andela/Ah-backend-xmen
# #162725273 Users report articles

#### What does this PR do?
Adds functionality to enable authenticated users to report on articles that violate the terms of authors heaven to the admins for further action or removal from the platform.

#### Description of the task to be implemented.
- Create a model to store reported articles
- Implement logic to inform admins when an article is reported
- Implement logic to inform a user that a particular article has been reported and action is being taken
- Implement logic to prevent an author from reporting their own articles.

#### How should this be manually tested?
- Fetch this branch by running this command `git fetch origin ft-report-articles-162725273`
- Check into this branch by running `git checkout ft-report-articles-162725273`
- Run migrations by running `python manage.py migrate`
- Start the development server by running `python manage.py runserver`
- login into the applications by making a **POST** to `/api/users/login/`
- create an article and log out.
- log back in using login credentials different from article author.
- Make a **POST** request to `/api/articles/<slug>/report`
- Expects a user token of a user different from the article author
- Log into application email account to see reported articles.
- Sample request body
` {
     "reason": "reason why you're reporting article"
}`

#### Screenshots
- Author trying to report their own article fails
<img width="1136" alt="screenshot 2019-01-29 at 17 19 08" src="https://user-images.githubusercontent.com/18280598/51914605-51034900-23ea-11e9-98a0-5a7b411549f2.png">

- Logged in user reporting article succeeds
<img width="1117" alt="screenshot 2019-01-29 at 17 24 01" src="https://user-images.githubusercontent.com/18280598/51914854-cec75480-23ea-11e9-9dcf-ef33eca1c062.png">



#### Related Pivotal Tracker stories.
[#162725273](https://www.pivotaltracker.com/story/show/162725273)

[[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/390fed78d0c49482d1b4)
